src/tcn_ext/tcn_keys.rs

```diff
@@ -1,80 +1,169 @@
+            .expect("Couldn't write TCK bytes");
+        Self::byte_vec_to_tck_byte_array(buf)
+    }
+
+    fn bytes_to_tck(tck: TckBytesWrapper) -> TemporaryContactKey {
+        TemporaryContactKey::read(Cursor::new(&tck)).expect("Couldn't read TCK bytes")
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_rak() {
+        let new_key = ReportAuthorizationKey::new(rand::thread_rng());
+        // println!("{}", new_key);
+
+        let bytes = TcnKeysImpl::<PreferencesMock>::rak_to_bytes(new_key);
+        println!("{:?}", bytes);
+        assert_eq!(1, 1);
+    }
+
+    #[test]
+    fn test_load_rak() {
+        let bytes = [
+            42, 118, 64, 131, 236, 36, 122, 23, 13, 108, 73, 171, 102, 145, 66, 91, 157, 105, 195,
+            126, 139, 162, 15, 31, 0, 22, 31, 230, 242, 241, 225, 85,
+        ];
+        let key = TcnKeysImpl::<PreferencesMock>::bytes_to_rak(bytes);
+
+        let tck = key.initial_temporary_contact_key();
+
+        println!("tck initial: {:#?}", tck);
+
+        let tck_bytes = TcnKeysImpl::<PreferencesMock>::tck_to_bytes(tck);
+
+        println!("Bytes: {:#?}", tck_bytes);
+    }
+
+    #[test]
+    fn test_load_tck() {
+        let rak_bytes = [
+            42, 118, 64, 131, 236, 36, 122, 23, 13, 108, 73, 171, 102, 145, 66, 91, 157, 105, 195,
+            126, 139, 162, 15, 31, 0, 22, 31, 230, 242, 241, 225, 85,
+        ];
+        let rak = TcnKeysImpl::<PreferencesMock>::bytes_to_rak(rak_bytes);
+        let tck_1 = rak.initial_temporary_contact_key();
+
+        let tck_inner_bytes = [
+            34, 166, 47, 23, 224, 52, 240, 95, 140, 186, 95, 243, 26, 13, 174, 128, 224, 229, 158,
+            248, 117, 7, 118, 110, 108, 57, 67, 206, 129, 22, 84, 13,
+        ];
+        println!("count = {}", tck_inner_bytes.len());
+
+        let version_bytes: [u8; 2] = [1, 0];
+
+        let version_vec = version_bytes.to_vec();
+        let rak_vec = rak_bytes.to_vec();
+
+        let tck_inner_vec = tck_inner_bytes.to_vec();
+
+        let complete_tck_vec = [&version_vec[..], &rak_vec[..], &tck_inner_vec[..]].concat();
+
+        let tck_bytes_wrapped =
+            TcnKeysImpl::<PreferencesMock>::byte_vec_to_tck_byte_array(complete_tck_vec);
+
+        let tck = TcnKeysImpl::<PreferencesMock>::bytes_to_tck(tck_bytes_wrapped);
+
+        println!("{:#?}", tck);
+    }
+
+    #[test]
+    fn test_generate_tcns() {
+        let rak_bytes = [
+            42, 118, 64, 131, 236, 36, 122, 23, 13, 108, 73, 171, 102, 145, 66, 91, 157, 105, 195,
+            126, 139, 162, 15, 31, 0, 22, 31, 230, 242, 241, 225, 85,
+        ];
+        let rak = TcnKeysImpl::<PreferencesMock>::bytes_to_rak(rak_bytes);
+        let mut tck = rak.initial_temporary_contact_key(); // tck <- tck_1
+        let mut tcns = Vec::new();
+
+        for _ in 0..100 {
+            tcns.push(tck.temporary_contact_number());
+            tck = tck.ratchet().unwrap();
+        }
```

Comment:
There's no test?

Reply:
Hmm, these tests aren't ideal, as it's not easy to see what exactly they are testing. E.g. here, are we testing that `unwrap` doesn't crash? Why specifically 100? Etc. But they're helpful for early stages development so we can keep them for now.